### ISSUE TITLE
Customisation of minimisers

### ISSUE THREAD

**Issue by apuignav** [zfit/zfit#84]
For example, in https://github.com/zfit/zfit/blob/develop/zfit/minimizers/minimizer_minuit.py#L78

We need to be able to pass parameters to configure the minimisers.

**Comment by mayou36** [comment]
Yes, absolutely.

Two different distinctions:
---------------------------

*general attributes*

General attributes for *all minimizers* that are given in the `__init__` (and a direct `minimizer.setter`):

- name
- verbosity (from 0 to 10, but not every step has to be supported?)

- a general strategy flag? (like a number from 0-10, each minimizer can do whatever he want's with that

General attributes for *specific minimizers*:

- configurations that are minimizer specific, like `learning_rate`. They should probably not be directly accesible but rather through `minimizer.config['attr']` so we can use the "exact" same name as in the actual minimizer without having any name clash


*minimization arguments*

the distinction between general and minimization arguments is non-trivial... But rather more in general (they can always be changed, but it means there's no need to pass around dicts with arguments for minimize)

for *all minimizers*:

- loss
- params

for *specific minimizers*:
- a dict with args, like: minimization_kwargs, that is passed. (Why not just kwargs: keeps the API cleaner, minimzie takes three arguments. Always. For the instanciation a different API for each minimizers init makes more sense (like: each pdf has a different init but the same `.integrate` or `.pdf` function. At least my current tendency)

**Comment by mayou36** [comment]
Issue moved to [zfit/zfit-development #33](https://github.com/zfit/zfit-development/issues/33) via [**ZenHub**](https://www.zenhub.com/)
